src/connector/mssql.rs

```diff
@@ -87,56 +88,24 @@ pub(crate) struct MssqlQueryParams {
     max_idle_connection_lifetime: Option<Duration>,
 }
 
-#[derive(Debug, Clone, Copy)]
-/// Controls the locking and row versioning behavior of Transact-SQL statements
-/// issued by a connection to SQL Server. Read more from the [SQL Server
-/// documentation].
-///
-/// [SQL Server documentation]: https://docs.microsoft.com/en-us/sql/t-sql/statements/set-transaction-isolation-level-transact-sql?view=sql-server-ver15
-#[cfg_attr(feature = "docs", doc(cfg(feature = "mssql")))]
-pub enum IsolationLevel {
-    ReadUncommitted,
-    ReadCommitted,
-    RepeatableRead,
-    Snapshot,
-    Serializable,
-}
-
-impl fmt::Display for IsolationLevel {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match self {
-            Self::ReadUncommitted => write!(f, "READ UNCOMMITTED"),
-            Self::ReadCommitted => write!(f, "READ COMMITTED"),
-            Self::RepeatableRead => write!(f, "REPEATABLE READ"),
-            Self::Snapshot => write!(f, "SNAPSHOT"),
-            Self::Serializable => write!(f, "SERIALIZABLE"),
-        }
-    }
-}
-
-impl FromStr for IsolationLevel {
-    type Err = Error;
-
-    fn from_str(s: &str) -> crate::Result<Self> {
-        match s {
-            "READ UNCOMMITTED" => Ok(Self::ReadUncommitted),
-            "READ COMMITTED" => Ok(Self::ReadCommitted),
-            "REPEATABLE READ" => Ok(Self::RepeatableRead),
-            "SNAPSHOT" => Ok(Self::Snapshot),
-            "SERIALIZABLE" => Ok(Self::Serializable),
-            _ => {
-                let kind = ErrorKind::database_url_is_invalid(format!("Invalid isolation level `{}`", s));
-
-                Err(Error::builder(kind).build())
-            }
-        }
-    }
-}
+static SQL_SERVER_DEFAULT_ISOLATION: IsolationLevel = IsolationLevel::ReadCommitted;
 
 #[async_trait]
 impl TransactionCapable for Mssql {
-    async fn start_transaction(&self) -> crate::Result<Transaction<'_>> {
-        Transaction::new(self, "BEGIN TRAN").await
+    async fn start_transaction(&self, isolation: Option<IsolationLevel>) -> crate::Result<Transaction<'_>> {
+        // Isolation levels in SQL Server are set on the connection and live until they're changed.
+        // Always explicitly setting the isolation level each time a tx is started (either to the given value
+        // or by using the default/connection string value) prevents transactions started on connections from
+        // the pool to have unexpected isolation levels set.
+        Transaction::new(
```

Comment:
I would change this to something like:
```
Transaction::new(
  self,
  "BEGIN TRAN",
  self.get_isolation_level(isolation)
);

```
That way you can put all the isolation comments in that function.